radixdlt-java/src/main/java/com/radixdlt/client/core/network/jsonrpc/RadixJsonRpcClient.java

```diff
@@ -62,106 +82,63 @@
-		return wsClient.getStatus();
-	}
-
-	/**
-	 * Attempts to close the websocket this client is connected to.
-	 * If there are still observers connected to the websocket closing
-	 * will not occur.
-	 *
-	 * @return true if websocket was successfully closed, false otherwise
-	 */
-	protected boolean tryClose() {
-		return this.wsClient.close();
-	}
-
 	/**
 	 * Generic helper method for calling a JSON-RPC method. Deserializes the received json.
 	 *
 	 * @param method name of JSON-RPC method
 	 * @return response from rpc method
 	 */
-	private Single<JsonElement> jsonRpcCall(String method, JsonObject params) {
-		return this.wsClient.connect().andThen(
-			Single.<JsonElement>create(emitter -> {
-				final String uuid = UUID.randomUUID().toString();
-
-				final JsonObject requestObject = new JsonObject();
-				requestObject.addProperty("id", uuid);
-				requestObject.addProperty("method", method);
-				requestObject.add("params", params);
-
-				Disposable d = messages
-					.filter(msg -> msg.has("id"))
-					.filter(msg -> msg.get("id").getAsString().equals(uuid))
-					.firstOrError()
-					.doOnSubscribe(disposable -> {
-						boolean sendSuccess = wsClient.send(GsonJson.getInstance().stringFromGson(requestObject));
-						if (!sendSuccess) {
-							disposable.dispose();
-							emitter.onError(new RuntimeException("Could not connect."));
-						}
-					})
-					.subscribe(
-						msg -> {
-							final JsonObject received = msg.getAsJsonObject();
-							if (received.has("result")) {
-								emitter.onSuccess(received.get("result"));
-							} else if (received.has("error")) {
-								emitter.onError(new JsonRpcException(requestObject, received));
-							} else {
-								emitter.onError(
-									new RuntimeException("Received bad json rpc message: " + received.toString())
-								);
-							}
-						},
-						err -> {
-							emitter.onError(new RuntimeException(err.getMessage()));
-						}
-					);
-
-				emitter.setCancellable(() -> {
-					d.dispose();
-					this.wsClient.close();
-				});
-			})
-		);
+	private Single<JsonRpcResponse> jsonRpcCall(String method, JsonObject params) {
+		return Single.create(emitter -> {
+			final String uuid = UUID.randomUUID().toString();
+
+			final JsonObject requestObject = new JsonObject();
+			requestObject.addProperty("id", uuid);
+			requestObject.addProperty("method", method);
+			requestObject.add("params", params);
+
+			Disposable d = messages
+				.filter(msg -> msg.has("id"))
+				.filter(msg -> msg.get("id").getAsString().equals(uuid))
+				.firstOrError()
+				.map(msg -> {
+					final JsonObject jsonResponse = msg.getAsJsonObject();
+					return new JsonRpcResponse(!jsonResponse.has("error"), jsonResponse);
+				})
+				.subscribe(emitter::onSuccess);
+
+			boolean sendSuccess = channel.sendMessage(GsonJson.getInstance().stringFromGson(requestObject));
+			if (!sendSuccess) {
+				emitter.onError(new RuntimeException("Could not send message: " + method + " " + params));
```

Comment:
Is the error information from emitter discarded here? Is this intentional?

Reply:
Good catch

Reply:
wait nvm bad catch, there is no error information as it's just a call